core/win32/syscall.c

```diff
@@ -1723,6 +1727,9 @@ presys_TerminateProcess(dcontext_t *dcontext, reg_t *param_base)
          * the syscall, safest to do syscall right here rather than going
          * back to handle_system_call()
          */
+        /* i#2346: delay client threads termination by using synch-all instead of
```

Comment:
Comment is again confusing: makes it seem like this is what we're doing today: needs XXX and "we should..." or sthg